lighthouse-core/lib/page-functions.js

```diff
@@ -225,6 +225,59 @@ function getNodeSelector(node) {
   return parts.join(' > ');
 }
 
+/**
+ * This function checks if an element or an ancestor of an element is `position:fixed`.
+ * In addition we ensure that the element is capable of behaving as a `position:fixed`
+ * element, checking that it lives within a scrollable ancestor.
+ * @param {HTMLElement} element
+ * @return {boolean}
+ */
+/* istanbul ignore next */
+function isPositionFixed(element) {
+  /**
+   * @param {HTMLElement} element
+   * @return {boolean}
+   */
+  function hasScrollableAncestor(element) {
+    let currentEl = element.parentElement;
+    while (currentEl) {
+      if (currentEl.scrollHeight > currentEl.clientHeight) {
+        const yScroll = getStyleAttrValue(currentEl, 'overflowY');
+        // In case where we reach outermost element (`<html>`), default computed value of visible
+        // will be scrollable.
+        if ((yScroll === 'scroll' || yScroll === 'auto') ||
+        (yScroll === 'visible' && currentEl.parentElement === null)
+        ) {
+          return true;
+        }
+      }
+      currentEl = currentEl.parentElement;
+    }
+    return false;
+  }
+
+  /**
+   * @param {HTMLElement} element
+   * @param {string} attr
+   * @return {string}
+   */
+  function getStyleAttrValue(element, attr) {
+    // Check style before computedStyle as computedStyle is expensive.
+    return element.style[attr] || window.getComputedStyle(element)[attr];
+  }
+
+  let currentEl = element;
+  while (currentEl) {
+    const position = getStyleAttrValue(currentEl, 'position');
+    // Only truly fixed if an ancestor is scrollable.
+    if ((position === 'fixed' || position === 'sticky') && hasScrollableAncestor(currentEl)) {
```

Comment:
we should be able to return `false` early if `hasScrollableAncestor` is falsy, right? could probably do that outside the loop entirely